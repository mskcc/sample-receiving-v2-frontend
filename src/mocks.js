
export const initialFormStateMock = {
  allContainers: [],
  allApplications: [],
  allMaterials: [],
  allPatientIdFormats: [],
  applications: [],
  containers: [],
  formIsLoading: true,
  filteredContainers: [
    { id: 'Plates', value: 'Plates' },
    { id: 'Micronic Barcoded Tubes', value: 'Micronic Barcoded Tubes' },
  ],
  filteredContainersBS: [
    { id: 'Blocks/Slides/Tubes', value: 'Blocks/Slides/Tubes' },
  ],
  materials: [],
  picklists: {},
  selectedApplication: '',
  selectedMaterial: '',
  species: [],
  patientIDTypeNeedsFormatting: false,
}

export const initialGridStateMock = {
  columns: [],
  rows: [],
  error: '',
  form: [],
  gridIsLoading: false,
  nothingToChange: false,
}

export const filledGridStateMock = {
  upload: {
    grid: {
      columns: ['Block/Slide/TubeID', 'Sample ID'],
      rows: [
        { tubeId: '', userId: '' },
        { tubeId: '', userId: '' },
        { tubeId: '', userId: '' },
        { tubeId: '', userId: '' },
      ],
      error: '',
      form: {
        material: 'Tissue',
        application: 'CustomCapture',
        igo_request_id: '444444',
        number_of_samples: '4',
        species: 'Tuberculosis',
        container: 'Plates',
        patient_id_type: '',
      },
      gridIsLoading: false,
      nothingToChange: false,
      columnFeatures: [
        {
          columnHeader: 'Block/Slide/TubeID',
          data: 'tubeId',
        },
        {
          columnHeader: 'Sample ID',
          data: 'userId',
        },
      ],
    },
  },
}

export const initialFullStateMock = {
  upload: {
    form: {
      allContainers: [],
      allApplications: [],
      allMaterials: [],
      allPatientIdFormats: [],
      applications: [],
      containers: [],
      formIsLoading: false,
      filteredContainers: [
        { id: 'Plates', value: 'Plates' },
        { id: 'Micronic Barcoded Tubes', value: 'Micronic Barcoded Tubes' },
      ],
      filteredContainersBS: [
        { id: 'Blocks/Slides/Tubes', value: 'Blocks/Slides/Tubes' },
      ],
      materials: [],
      selectedApplication: '',
      selectedMaterial: '',
      species: [],
      patientIDTypeNeedsFormatting: false,
    },
    grid: {
      columns: [],
      rows: [],
      error: '',
      form: [],
      gridIsLoading: false,
      nothingToChange: false,
    },
  },
}

export const filledFullStateMock = {
  upload: {
    form: {
      allContainers: [{ id: 'test', value: 'test' }],
      allApplications: [{ id: 'test', value: 'test' }],
      allMaterials: [{ id: 'test', value: 'test' }],
      allPatientIdFormats: [{ id: 'test', value: 'test' }],
      applications: [{ id: 'test', value: 'test' }],
      containers: [{ id: 'test', value: 'test' }],
      formIsLoading: false,
      filteredContainers: [
        { id: 'Plates', value: 'Plates' },
        { id: 'Micronic Barcoded Tubes', value: 'Micronic Barcoded Tubes' },
      ],
      filteredContainersBS: [
        { id: 'Blocks/Slides/Tubes', value: 'Blocks/Slides/Tubes' },
      ],
      materials: [{ id: 'test', value: 'test' }],
      selectedApplication: '',
      selectedMaterial: '',
      species: [{ id: 'test', value: 'test' }],
      patientIDTypeNeedsFormatting: false,
    },
    grid: {
      columns: ['Block/Slide/TubeID', 'Sample ID'],
      rows: [{ tubeId: '', userId: '' }, { tubeId: '', userId: '' }],
      error: '',
      form: {
        material: 'Tissue',
        application: 'CustomCapture',
        igo_request_id: '444444',
        number_of_samples: '4',
        species: 'Tuberculosis',
        container: 'Plates',
        patient_id_type: '',
      },
      gridIsLoading: false,
      nothingToChange: false,
      columnFeatures: [
        {
          columnHeader: 'Block/Slide/TubeID',
          data: 'tubeId',

          name: 'Block/Slide/TubeID',
        },
        {
          columnHeader: 'Sample ID',
          data: 'userId',
        },
      ],
    },
  },
}

export const formValuesMock = {
  application: 'CustomCapture',
  container: 'Plates',
  igo_request_id: '444444',
  material: 'Tissue',
  number_of_samples: '2',
  patient_id_type: '',
  species: 'Tuberculosis',
}

export const columnDefsResponseMock = {
  columnDefs: [
    {
      columnHeader: 'Block/Slide/TubeID',
      data: 'tubeId',
      editableCellTemplate:
        '<div><form name="inputForm"><input class="form-control inputheight" type="INPUT_TYPE" ng-class="\'colt\' + col.uid" ui-grid-editor ng-model="MODEL_COL_FIELD" ng-paste="grid.appScope.handleCellPaste($event)"></form></div>',
      headerCellClass: 'required',
      helpText:
        'The identifier on your tube, block or slide.  You can paste in directly from excel, and there are no formatting rules.  Please be as correct as possible, and ensure your tubes, blocks and slides are labeled clearly.',
      name: 'Block/Slide/TubeID',
    },
    {
      columnHeader: 'Sample ID',
      data: 'userId',
      editableCellTemplate:
        '<div><form name="inputForm"><input class="form-control inputheight" type="INPUT_TYPE" ng-class="\'colt\' + col.uid" ui-grid-editor ng-model="MODEL_COL_FIELD" ng-paste="grid.appScope.handleCellPaste($event)"></form></div>',
      headerCellClass: 'required',
      helpText:
        'The Sample ID stays with your sample for its lifetime. Letters, numbers, dashes, and underscores only, three char min. You cannot have more than one underscore consecutively.',
      name: 'Sample ID',
    },
    {
      columnHeader: 'Species',
      data: 'organism',
      editableCellTemplate: 'ui-grid/dropdownEditor',
      headerCellClass: 'required',
      helpText:
        'If your species is not available, please contact IGO immediately',
      name: 'Species',
      picklistName: 'Species',
      source: [
        { id: 'Human', value: 'Human' },
        { id: 'Mouse', value: 'Mouse' },
        { id: 'Mouse_GeneticallyModified', value: 'Mouse_GeneticallyModified' },
        { id: 'Drosophilia', value: 'Drosophilia' },
        { id: 'Zebrafish', value: 'Zebrafish' },
        { id: 'Chicken', value: 'Chicken' },
        { id: 'Bacteria', value: 'Bacteria' },
        { id: 'S.Cerevisae', value: 'S.Cerevisae' },
        { id: 'Tuberculosis', value: 'Tuberculosis' },
        { id: 'E.Coli ', value: 'E.Coli ' },
        { id: 'C.Elegans', value: 'C.Elegans' },
        { id: 'S.Pombe', value: 'S.Pombe' },
        { id: 'R.norvegicus', value: 'R.norvegicus' },
        { id: 'R.rattus', value: 'R.rattus' },
        { id: 'Plasmid', value: 'Plasmid' },
        { id: 'Archaea', value: 'Archaea' },
        { id: 'other', value: 'other' },
      ],
      type: 'dropdown',
      trimDropdown: false,
    },
  ],
}

export const gridMock = {
  columnFeatures: [
    {
      columnHeader: 'Block/Slide/TubeID',
      data: 'tubeId',
      editableCellTemplate: `<div><form name="inputForm"><input class="form-control inputheight" type="INPUT_TYPE" ng-class="'colt' + col.uid" ui-grid-editor ng-model="MODEL_COL_FIELD" ng-paste="grid.appScope.handleCellPaste($event)"></form></div>`,
      headerCellClass: 'required',
      helpText:
        'The identifier on your tube, block or slide.  You can paste in directly from excel, and there are no formatting rules.  Please be as correct as possible, and ensure your tubes, blocks and slides are labeled clearly.',
      name: 'Block/Slide/TubeID',
    },
    {
      columnHeader: 'Sample ID',
      data: 'userId',
      editableCellTemplate: `<div><form name="inputForm"><input class="form-control inputheight" type="INPUT_TYPE" ng-class="'colt' + col.uid" ui-grid-editor ng-model="MODEL_COL_FIELD" ng-paste="grid.appScope.handleCellPaste($event)"></form></div>`,
      headerCellClass: 'required',
      helpText:
        'The Sample ID stays with your sample for its lifetime. Letters, numbers, dashes, and underscores only, three char min. You cannot have more than one underscore consecutively.',
      name: 'Sample ID',
    },
    {
      columnHeader: 'Species',
      data: 'organism',
      editableCellTemplate: 'ui-grid/dropdownEditor',
      headerCellClass: 'required',
      helpText:
        'If your species is not available, please contact IGO immediately',
      name: 'Species',
      picklistName: 'Species',
      source: [
        'Human',
        'Mouse',
        'Mouse_GeneticallyModified',
        'Drosophilia',
        'Zebrafish',
        'Chicken',
        'Bacteria',
        'S.Cerevisae',
        'Tuberculosis',
        'E.Coli ',
        'C.Elegans',
        'S.Pombe',
        'R.norvegicus',
        'R.rattus',
        'Plasmid',
        'Archaea',
        'other',
      ],
      type: 'dropdown',
      trimDropdown: false,
    },
  ],
  columnHeaders: ['Block/Slide/TubeID', 'Sample ID', 'Species'],
  rows: [
    {
      tubeId: '',
      userId: '',
      organism: 'Tuberculosis',
    },
    {
      tubeId: '',
      userId: '',
      organism: 'Tuberculosis',
    },
  ],
}
