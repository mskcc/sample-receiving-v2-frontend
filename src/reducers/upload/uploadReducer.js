import { combineReducers } from 'redux'

import uploadFormReducer from './uploadFormReducer'
import uploadGridReducer from './uploadGridReducer'

export default combineReducers({
  form: uploadFormReducer,
  grid: uploadGridReducer,
})
