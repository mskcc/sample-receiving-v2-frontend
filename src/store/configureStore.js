import { createStore, applyMiddleware, compose } from 'redux'
import { Provider } from 'react-redux'

import thunk from 'redux-thunk'
import multi from 'redux-multi'
import { persistStore } from 'redux-persist'
import { createLogger } from 'redux-logger'
import rootReducer from '../reducers/rootReducer'
import DevTools from '../containers/DevTools'

const middleware = [thunk, multi]
if (process.env.NODE_ENV === 'production') {
}

const store = createStore(
  rootReducer,
  compose(
    applyMiddleware(...middleware),
    DevTools.instrument()
  )
)


let persistor = persistStore(store)

export {store, persistor}
