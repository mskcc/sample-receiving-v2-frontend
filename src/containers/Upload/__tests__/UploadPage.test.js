// check if grid length changes after sample number was changed
// check if grid is generated
import React, { Component } from 'react'

import { applyMiddleware, createStore } from 'redux'
import thunk from 'redux-thunk'
import multi from 'redux-multi'
import moxios from 'moxios'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'

import renderer from 'react-test-renderer'
import { mount } from 'enzyme'

import uploadReducer from '../../../reducers/upload/uploadReducer'

import TestUploadPage from '../UploadPage'
import uploadFormReducer from '../../../reducers/upload/uploadFormReducer'
// import uploadGridActions from '../../../actions/upload/uploadGridActions'

import { initialFullStateMock, filledFullStateMock } from '../../../mocks'

const middlewares = [thunk, multi]
const mockStore = configureStore(middlewares)

const flushAllPromises = () => new Promise(resolve => setImmediate(resolve))
const UploadPage = props => (
  <Provider store={props.store}>
    <TestUploadPage />
  </Provider>
)

// TODO first few are basically unit tests, should be moved eventually
describe('renders UploadPage', () => {
  beforeEach(function() {
    moxios.install()
  })

  afterEach(function() {
    moxios.uninstall()
  })

  it('renders itself and UploadForm correctly', () => {
    const UploadFormContainerComponent = renderer
      .create(<UploadPage store={mockStore(initialFullStateMock)} />)
      .toJSON()
    expect(UploadFormContainerComponent).toMatchSnapshot()
  })

  it('renders UploadForm child', async () => {
    const store = mockStore(initialFullStateMock)
    const wrapper = mount(
      <UploadPage store={mockStore(initialFullStateMock)} />
    )
    expect(wrapper.find('UploadForm').exists()).toBe(true)
  })
  it('renders UploadGrid', async () => {
    // const getColumns = jest.fn()
    const store = mockStore(filledFullStateMock)
    // let oldState = store.getState()
    // const uploadGridActions = { getColumns: jest.fn() }
    const wrapper = mount(
      <UploadPage
        store={store}
        // uploadGridActions={uploadGridActions}
        // getColumns={getColumns}
      />
    )

    // request initial state for grid on button click
    // wrapper
    //   .find('#upload-form')
    //   .first()
    //   .simulate('submit')
    // await flushAllPromises()
    // let newStore = store.getState()
    // console.log(oldState)
    // console.log(newStore)
    expect(wrapper.find('UploadGrid').exists()).toBe(true)

    // expect(newStore).not.toEqual(oldState)
    // console.log(wrapper.find('UploadForm').instance().state)

    // moxios.wait(() => {
    //   const request = moxios.requests.mostRecent()
    //   request.respondWith({
    //     status: 200,
    //     response: getChoicesForDNALibraryMock,
    //   })
    //   let actions = store.getActions()
    //   console.log(actions)
    // })
    // let actions = store.getActions()
    // console.log(actions)
    // await flushAllPromises()
    // wrapper.update()
    // expect(getColumns).toHaveBeenCalledTimes(1)
    // console.log(wrapper.find('UploadGrid').instance().state)
    // expect(actions).toEqual(expectedActions)
  })
})

// const UploadFormContainer = props => (
//   <Provider store={mockStore(initialFullStateMock)}>
//     <TestUploadFormContainer />
//   </Provider>
// )

// describe('Render UploadFormContainer', () => {
//   it('render UploadFormContainer correctly', () => {
//     const UploadFormContainerComponent = renderer
//       .create(<UploadFormContainer />)
//       .toJSON()
//     expect(UploadFormContainerComponent).toMatchSnapshot()
//   })
// })

// describe('responds to input', () => {
//   it('has species field', () => {
//     const wrapper = mount(<UploadFormContainer />)
//     expect(wrapper.find('#material').exists()).toBe(true)
//   })

// })

// const middlewares = [thunk, multi]
// const mockStore = configureStore(middlewares)

// describe('should handle combinations', () =>{
// it('filters containers on material changes', () => {

// })

// })
// })

// if species with formatter selected -> show -> validate
// if species without formatter selected -> show = false -> skip validate

// it('has species field', () => {
//   const onChangeMock = jest.fn()
//   const wrapper = mount(<UploadFormContainer />)
//   const materialInput = wrapper.find('#material').first()

//   const event = { target: { value: 'Cells' } }
//   materialInput.simulate('change', event)
//   console.log(store.getState())
//   console.log(materialInput.value)
//   expect(store.getState().upload.form.selectedMaterial).toEqual('Cells')
// })
