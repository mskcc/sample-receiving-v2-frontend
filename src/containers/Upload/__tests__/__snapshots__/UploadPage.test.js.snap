// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders UploadPage renders itself and UploadForm correctly 1`] = `
<div
  className="MuiPaper-root-25 MuiPaper-elevation1-28 MuiPaper-rounded-26 UploadForm-container-17"
>
  <form
    className="UploadForm-form-18"
    id="upload-form"
    onSubmit={[Function]}
  >
    <div
      className="Dropdown-textField-52"
    >
      <div
        aria-expanded={false}
        aria-haspopup="listbox"
        aria-labelledby="downshift-0-label"
        aria-owns={null}
        role="combobox"
      >
        <div
          className="MuiFormControl-root-53 MuiFormControl-fullWidth-56"
        >
          <label
            className="MuiFormLabel-root-68 MuiInputLabel-root-57 MuiInputLabel-formControl-62 MuiInputLabel-animated-65"
            data-shrink={false}
            htmlFor="downshift-0-input"
            id="downshift-0-label"
          >
            Missing translationId: upload.form.material_label for language: \${ languageCode }
          </label>
          <div
            aria-activedescendant={null}
            aria-autocomplete="list"
            aria-controls={null}
            aria-labelledby="downshift-0-label"
            className="MuiInputBase-root-88 MuiInput-root-75 MuiInput-underline-79 MuiInputBase-formControl-89 MuiInput-formControl-76 MuiInputBase-adornedEnd-93"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoComplete="off"
              className="MuiInputBase-input-98 MuiInput-input-83 MuiInputBase-inputAdornedEnd-104"
              disabled={false}
              id="material"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              required={false}
              type="text"
              value=""
            />
            <div
              className="MuiInputAdornment-root-105 MuiInputAdornment-positionEnd-108"
            >
              <button
                aria-label="Toggle menu open"
                className="MuiButtonBase-root-116 MuiIconButton-root-110"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onContextMenu={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex="0"
                type="button"
              >
                <span
                  className="MuiIconButton-label-115"
                >
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root-119"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                  </svg>
                </span>
                <span
                  className="MuiTouchRipple-root-225"
                />
              </button>
            </div>
          </div>
          <p
            className="MuiFormHelperText-root-128"
          >
            Missing translationId: upload.form.material_helptext for language: \${ languageCode } (0 choices)
          </p>
        </div>
      </div>
    </div>
    <div
      className="Dropdown-textField-52"
    >
      <div
        aria-expanded={false}
        aria-haspopup="listbox"
        aria-labelledby="downshift-1-label"
        aria-owns={null}
        role="combobox"
      >
        <div
          className="MuiFormControl-root-53 MuiFormControl-fullWidth-56"
        >
          <label
            className="MuiFormLabel-root-68 MuiInputLabel-root-57 MuiInputLabel-formControl-62 MuiInputLabel-animated-65"
            data-shrink={false}
            htmlFor="downshift-1-input"
            id="downshift-1-label"
          >
            Missing translationId: upload.form.application_label for language: \${ languageCode }
          </label>
          <div
            aria-activedescendant={null}
            aria-autocomplete="list"
            aria-controls={null}
            aria-labelledby="downshift-1-label"
            className="MuiInputBase-root-88 MuiInput-root-75 MuiInput-underline-79 MuiInputBase-formControl-89 MuiInput-formControl-76 MuiInputBase-adornedEnd-93"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoComplete="off"
              className="MuiInputBase-input-98 MuiInput-input-83 MuiInputBase-inputAdornedEnd-104"
              disabled={false}
              id="application"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              required={false}
              type="text"
              value=""
            />
            <div
              className="MuiInputAdornment-root-105 MuiInputAdornment-positionEnd-108"
            >
              <button
                aria-label="Toggle menu open"
                className="MuiButtonBase-root-116 MuiIconButton-root-110"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onContextMenu={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex="0"
                type="button"
              >
                <span
                  className="MuiIconButton-label-115"
                >
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root-119"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                  </svg>
                </span>
                <span
                  className="MuiTouchRipple-root-225"
                />
              </button>
            </div>
          </div>
          <p
            className="MuiFormHelperText-root-128"
          >
            Missing translationId: upload.form.application_helptext for language: \${ languageCode } (0 choices)
          </p>
        </div>
      </div>
    </div>
    <fieldset
      className="MuiFormControl-root-53"
    >
      <div
        className="Dropdown-textField-52"
      >
        <div
          aria-expanded={false}
          aria-haspopup="listbox"
          aria-labelledby="downshift-2-label"
          aria-owns={null}
          role="combobox"
        >
          <div
            className="MuiFormControl-root-53 MuiFormControl-fullWidth-56"
          >
            <label
              className="MuiFormLabel-root-68 MuiInputLabel-root-57 MuiInputLabel-formControl-62 MuiInputLabel-animated-65"
              data-shrink={false}
              htmlFor="downshift-2-input"
              id="downshift-2-label"
            >
              Missing translationId: upload.form.species_label for language: \${ languageCode }
            </label>
            <div
              aria-activedescendant={null}
              aria-autocomplete="list"
              aria-controls={null}
              aria-labelledby="downshift-2-label"
              className="MuiInputBase-root-88 MuiInput-root-75 MuiInput-underline-79 MuiInputBase-formControl-89 MuiInput-formControl-76 MuiInputBase-adornedEnd-93"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoComplete="off"
                className="MuiInputBase-input-98 MuiInput-input-83 MuiInputBase-inputAdornedEnd-104"
                disabled={false}
                id="species"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                required={false}
                type="text"
                value=""
              />
              <div
                className="MuiInputAdornment-root-105 MuiInputAdornment-positionEnd-108"
              >
                <button
                  aria-label="Toggle menu open"
                  className="MuiButtonBase-root-116 MuiIconButton-root-110"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onContextMenu={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex="0"
                  type="button"
                >
                  <span
                    className="MuiIconButton-label-115"
                  >
                    <svg
                      aria-hidden="true"
                      className="MuiSvgIcon-root-119"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M7 10l5 5 5-5z"
                      />
                      <path
                        d="M0 0h24v24H0z"
                        fill="none"
                      />
                    </svg>
                  </span>
                  <span
                    className="MuiTouchRipple-root-225"
                  />
                </button>
              </div>
            </div>
            <p
              className="MuiFormHelperText-root-128"
            >
              Missing translationId: upload.form.species_helptext for language: \${ languageCode } (0 choices)
            </p>
          </div>
        </div>
      </div>
    </fieldset>
    <div
      className="Dropdown-textField-52"
    >
      <div
        aria-expanded={false}
        aria-haspopup="listbox"
        aria-labelledby="downshift-3-label"
        aria-owns={null}
        role="combobox"
      >
        <div
          className="MuiFormControl-root-53 MuiFormControl-fullWidth-56"
        >
          <label
            className="MuiFormLabel-root-68 MuiInputLabel-root-57 MuiInputLabel-formControl-62 MuiInputLabel-animated-65"
            data-shrink={false}
            htmlFor="downshift-3-input"
            id="downshift-3-label"
          >
            Missing translationId: upload.form.container_label for language: \${ languageCode }
          </label>
          <div
            aria-activedescendant={null}
            aria-autocomplete="list"
            aria-controls={null}
            aria-labelledby="downshift-3-label"
            className="MuiInputBase-root-88 MuiInput-root-75 MuiInput-underline-79 MuiInputBase-formControl-89 MuiInput-formControl-76 MuiInputBase-adornedEnd-93"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoComplete="off"
              className="MuiInputBase-input-98 MuiInput-input-83 MuiInputBase-inputAdornedEnd-104"
              disabled={false}
              id="container"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              required={false}
              type="text"
              value=""
            />
            <div
              className="MuiInputAdornment-root-105 MuiInputAdornment-positionEnd-108"
            >
              <button
                aria-label="Toggle menu open"
                className="MuiButtonBase-root-116 MuiIconButton-root-110"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onContextMenu={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex="0"
                type="button"
              >
                <span
                  className="MuiIconButton-label-115"
                >
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root-119"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                  </svg>
                </span>
                <span
                  className="MuiTouchRipple-root-225"
                />
              </button>
            </div>
          </div>
          <p
            className="MuiFormHelperText-root-128"
          >
            Missing translationId: upload.form.container_helptext for language: \${ languageCode }
          </p>
        </div>
      </div>
    </div>
    <div
      className="MuiFormControl-root-53 Input-textField-137"
    >
      <label
        className="MuiFormLabel-root-68 MuiInputLabel-root-57 MuiInputLabel-formControl-62 MuiInputLabel-animated-65"
        data-shrink={false}
        htmlFor="number_of_samples"
      >
        Missing translationId: upload.form.number_of_samples_label for language: \${ languageCode }
      </label>
      <div
        className="MuiInputBase-root-88 MuiInput-root-75 MuiInput-underline-79 MuiInputBase-formControl-89 MuiInput-formControl-76"
        onClick={[Function]}
      >
        <input
          aria-describedby="number_of_samples-helper-text"
          aria-invalid={false}
          className="MuiInputBase-input-98 MuiInput-input-83 MuiInputBase-inputType-101 MuiInput-inputType-86"
          disabled={false}
          id="number_of_samples"
          min={0}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="number"
        />
      </div>
      <p
        className="MuiFormHelperText-root-128"
        id="number_of_samples-helper-text"
      >
        Missing translationId: upload.form.number_of_samples_helptext for language: \${ languageCode }
      </p>
    </div>
    <fieldset
      className="MuiFormControl-root-53"
    >
      <div
        className="MuiFormControl-root-53 Input-textField-137"
      >
        <label
          className="MuiFormLabel-root-68 MuiFormLabel-filled-72 MuiInputLabel-root-57 MuiInputLabel-formControl-62 MuiInputLabel-animated-65 MuiInputLabel-shrink-64"
          data-shrink={true}
          htmlFor="igo_request_id"
        >
          Missing translationId: upload.form.igo_request_id_label for language: \${ languageCode }
        </label>
        <div
          className="MuiInputBase-root-88 MuiInput-root-75 MuiInput-underline-79 MuiInputBase-formControl-89 MuiInput-formControl-76 MuiInputBase-adornedStart-92"
          onClick={[Function]}
        >
          <div
            className="MuiInputAdornment-root-105 MuiInputAdornment-positionStart-107"
          >
            <p
              className="MuiTypography-root-138 MuiTypography-body1-147 MuiTypography-colorTextSecondary-171"
            >
              IGO-
            </p>
          </div>
          <input
            aria-describedby="igo_request_id-helper-text"
            aria-invalid={false}
            className="MuiInputBase-input-98 MuiInput-input-83 MuiInputBase-inputType-101 MuiInput-inputType-86 MuiInputBase-inputAdornedStart-103"
            disabled={false}
            id="igo_request_id"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            type="number"
            value="444444"
          />
        </div>
        <p
          className="MuiFormHelperText-root-128 MuiFormHelperText-filled-134"
          id="igo_request_id-helper-text"
        >
          Missing translationId: upload.form.igo_request_id_helptext for language: \${ languageCode }
        </p>
      </div>
      <div
        className="CheckboxComponent-container-174"
      >
        <label
          className="MuiFormControlLabel-root-177"
        >
          <span
            className="MuiButtonBase-root-116 MuiIconButton-root-110 MuiPrivateSwitchBase-root-189 MuiCheckbox-root-183 MuiCheckbox-colorSecondary-188"
            onBlur={[Function]}
            onContextMenu={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={null}
          >
            <span
              className="MuiIconButton-label-115"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root-119 MuiSvgIcon-fontSizeSmall-126"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
              </svg>
              <input
                checked={false}
                className="MuiPrivateSwitchBase-input-192"
                data-indeterminate={false}
                disabled={false}
                onChange={[Function]}
                type="checkbox"
                value="igo_alternative_id"
              />
            </span>
            <span
              className="MuiTouchRipple-root-225"
            />
          </span>
          <span
            className="MuiTypography-root-138 MuiTypography-body1-147 MuiFormControlLabel-label-182"
          >
            Missing translationId: upload.form.igo_alternative_id_label for language: \${ languageCode }
          </span>
        </label>
        <div>
          <div
            className="MuiPaper-root-25 MuiPaper-elevation1-28 MuiPaper-rounded-26 MuiCard-root-193 CheckboxComponent-card-175"
            style={
              Object {
                "opacity": 0,
              }
            }
          >
            <div
              className="MuiCardContent-root-194 CheckboxComponent-content-176"
            >
              <p
                className="MuiTypography-root-138 MuiTypography-body1-147 MuiTypography-colorTextSecondary-171"
              >
                Missing translationId: upload.form.igo_alternative_id_helptext for language: \${ languageCode }
                 
              </p>
            </div>
          </div>
        </div>
      </div>
    </fieldset>
  </form>
  <div
    className="Button-wrapper-196"
  >
    <button
      className="MuiButtonBase-root-116 MuiButton-root-199 MuiButton-contained-210 MuiButton-containedSecondary-212 MuiButton-raised-213 MuiButton-raisedSecondary-215 Button-button-195"
      disabled={false}
      form="upload-form"
      onBlur={[Function]}
      onContextMenu={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="submit"
    >
      <span
        className="MuiButton-label-200"
      >
        Missing translationId: upload.form_submit_label for language: \${ languageCode }
      </span>
      <span
        className="MuiTouchRipple-root-225"
      />
    </button>
    <div
      className="Button-nothingToSubmit-198"
      style={
        Object {
          "opacity": 0,
        }
      }
    >
      Missing translationId: upload.form.nothing_to_change for language: \${ languageCode }
    </div>
  </div>
   
</div>
`;
